src/printer.js

```diff
@@ -1608,6 +1613,21 @@ function printStatementSequence(path, options, print) {
   return join(hardline, printed);
 }
 
+function printPropertyKey(path, print) {
+  var node = path.getNode().key;
+  if (node.type === "StringLiteral") {
+    if (isIdentifierName(node.value)) { // 'a' -> a
+      return node.value;
+    }
+    let val = parseFloat(node.value);
+    let strVal = fromString(val);
+    if (val >= 0 && strVal === node.value) { // '0' -> 0
+      return strVal;
+    }
```

Comment:
I think we maybe should only do `"a" -> a` and not worry about numbers. It's very rare to actually use numbers in objects as keys, and I'd like to avoid having to maintain this code. I can tell it's the kind that I'd read later and have a hard time figuring out why we do this. There are also some perf concerns, and I don't want to die by a thousand cuts.

Can we just do:

```js
if(node.type === "StringLiteral" && isIdentifierName(node.value)) {
   return node.value; 
}
return path.call(print, "key");
```

?